src/webots/nodes/WbCamera.cpp

```diff
@@ -214,6 +231,20 @@ void WbCamera::initializeSharedMemory() {
   }
 }
 
+void WbCamera::initializeSegmentationSharedMemory() {
+  cCameraNumber++;
+  delete mSegmentationShm;
+  mSegmentationShm = initializeSharedMemory();
+  mHasSegmentationSharedMemoryChanged = true;
+  if (mSegmentationShm) {
+    unsigned char *data = (unsigned char *)mSegmentationShm->data();
+    // initialize the shared memory with a black image
+    int *im = reinterpret_cast<int *>(data);
+    for (int i = 0; i < width() * height(); i++)
```

Comment:
For performance, it would be better to cache `width() * height()`
```suggestion
    const size = width() * height();
    for (int i = 0; i < size; i++)
```

Reply:
Isn't GCC doing this kind of 'simple' optimization this for us?

Reply:
I think that we never measured the performance difference and in most of the cases we didn't store the size in a different variable (for example this is just a copy of what is already done in other classes for example WbLidar, etc.).

But I'm fine with this change.

Reply:
I believe gcc cannot guess what the `WbRenderingDevice.width()` method does and whether it will always return the same value, so it will repeatedly call this method (and the `height()` method as well), multiply their return values and compare with `i` at each loop step, which will certainly have a significant impact on the performance of such a simple loop. Depending on the image size, it can be quite big... I believe we should be aware of this when writing new code and probably, we should fix existing code when we spot the problem (e.g., in WbLidar, etc.).